crates/eth-rpc/tests/kkrt_prep.rs

```diff
@@ -0,0 +1,377 @@
+use std::collections::HashMap;
+use std::fs;
+use std::sync::Arc;
+
+use bytes::BytesMut;
+use dojo_test_utils::sequencer::TestSequencer;
+use ethers::abi::{Abi, Bytes as EthersBytes, JsonAbi};
+use reth_primitives::{
+    sign_message, Address, Bytes, Transaction, TransactionKind, TransactionSigned, TxEip1559, H256, U256,
+};
+use reth_rpc_types::CallRequest;
+use starknet::accounts::{Account, Call, ConnectedAccount, SingleOwnerAccount};
+use starknet::contract::ContractFactory;
+use starknet::core::chain_id;
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::{
+    BlockId, BlockTag, FieldElement, FunctionCall, InvokeTransactionReceipt, MaybePendingTransactionReceipt,
+    TransactionReceipt,
+};
+use starknet::core::utils::{get_contract_address, get_selector_from_name};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
+use url::Url;
+
+pub fn get_contract(filename: &str) -> (Abi, EthersBytes) {
+    let path = format!("./tests/compiled_solidity/{filename}");
+    let contents = fs::read_to_string(path).unwrap();
+    let obj: JsonAbi = serde_json::from_str(&contents).unwrap();
+    let JsonAbi::Object(obj) = obj else { panic!() };
+    (serde_json::from_str(&serde_json::to_string(&obj.abi).unwrap()).unwrap(), obj.bytecode.unwrap().to_vec())
```

Comment:
why is this serialization to deserialization (to switch types) necessary? I understand we have a obj of type `JsonAbi::Object(obj)`

Can't we access the `Abi` type from a JsonAbi::Object?

Reply:
i yoinked this as an example from ethers, will revisit to see if theres something cleaner